AppCenterCrashes/AppCenterCrashes/MSCrashes.mm

```diff
@@ -611,9 +618,15 @@ - (void)startDelayedCrashProcessing {
   // This must be performed asynchronously to prevent a deadlock with 'unprocessedCrashReports'.
   dispatch_time_t delay = dispatch_time(DISPATCH_TIME_NOW, (1 * NSEC_PER_SEC));
   dispatch_after(delay, dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-      [self startCrashProcessing];
+    [self startCrashProcessing];
 
-      dispatch_semaphore_signal(self.delayedProcessingSemaphore);
+    // Only release once to avoid releasing an unbounded number of times.
+    @synchronized(self) {
```

Comment:
A dispatch once should do the work nicely here instead of the synchronized + property:
```
  static dispatch_once_t onceToken;
  dispatch_once(&onceToken, ^{
    // Do something only once
  });
```

Reply:
Now that I think about it, that'd be better.

Reply:
I hesitate to use a `static` `onceToken`. The problem is that since this is an instance method with its own semaphore, any other instances will be deadlocked, which can be a problem for tests. So I guess I will need to `malloc` and `free` space for a `dispatch_token_t`? (If I make it a `@property`, then I run into the problem of needing to access its address, which can't be done.

Reply:
I'm hesitant to use `malloc`/`free`. I'm fine if we keep the variable around. It's easy to understand, which I like a lot.

Reply:
OK I am fine with that.